packages/aws-rfdk/lib/deadline/lib/sep-spotfleet.ts

```diff
@@ -0,0 +1,731 @@
+   */
+  readonly deadlineRegion?: string;
+
+  /**
+   * An IAM role to associate with the instance profile assigned to its resources.
+   *
+   * The role must be assumable by the service principal `ec2.amazonaws.com`,
+   * have AWSThinkboxDeadlineSpotEventPluginWorkerPolicy policy attached and
+   * the role name must begin with "DeadlineSpot":
+   *
+   *    const role = new iam.Role(this, 'MyRole', {
+   *      assumedBy: new iam.ServicePrincipal('ec2.amazonaws.com'),
+   *      managedPolicies: [
+   *        ManagedPolicy.fromAwsManagedPolicyName('AWSThinkboxDeadlineSpotEventPluginWorkerPolicy'),
+   *      ],
+   *      roleName: 'DeadlineSpot' + ...,
+   *    });
+   *
+   * @default - A role will automatically be created.
+   */
+  readonly role?: IRole;
+
+  /**
+   * Name of SSH keypair to grant access to instances.
+   *
+   * @default - No SSH access will be possible.
+   */
+  readonly keyName?: string;
+
+  /**
+   * Security Groups to assign to this fleet.
+   *
+   * @default - A new security group will be created automatically.
+   */
+  readonly securityGroups?: ISecurityGroup[];
+
+  /**
+   * User data that instances use when starting up.
+   *
+   * @default - User data will be created automatically.
+   */
+  readonly userData?: UserData;
+
+  /**
+   * The Block devices that will be attached to your workers.
+   *
+   * @default - The default devices of the provided ami will be used.
+   */
+  readonly blockDevices?: BlockDevice[];
+
+  /**
+   * Indicates how to allocate the target Spot Instance capacity
+   * across the Spot Instance pools specified by the Spot Fleet request.
+   *
+   * @default - SEPSpotFleetAllocationStrategy.LOWEST_PRICE.
+   */
+  readonly allocationStrategy?: SEPSpotFleetAllocationStrategy;
+
+  /**
+   * Where to place the instance within the VPC.
+   *
+   * @default - Private subnets.
+   */
+  readonly vpcSubnets?: SubnetSelection;
+
+  /**
+   * The end date and time of the request, in UTC format (YYYY -MM -DD T*HH* :MM :SS Z).
+   * After the end date and time, no new Spot Instance requests are placed or able to fulfill the request.
+   *
+   * @default - the Spot Fleet request remains until you cancel it.
+   */
+  readonly validUntil?: Expiration;
+
+  /**
+   * Properties for setting up the Deadline Worker's LogGroup
+   * @default - LogGroup will be created with all properties' default values and a prefix of "/renderfarm/".
+   */
+  readonly logGroupProps?: LogGroupFactoryProps;
+
+  /**
+   * An optional provider of user data commands to be injected at various points during the Worker configuration lifecycle.
+   * You can provide a subclass of InstanceUserDataProvider with the methods overridden as desired.
+   */
+  readonly userDataProvider?: IInstanceUserDataProvider;
+
+  /**
+   * The tags to apply during creation of instances.
+   *
+   * @default - Only RFDK tags are applied.
+   */
+  readonly instanceTags?: CfnTag[];
```

Comment:
Is it fine to use `CfnTag` type in this interface, or it's better to create my own similar type for tags?

Reply:
I think that we might actually be able to be a bit clever here and use the CDK's Tagging aspect.

Try something like:
```ts

public readonly tags: core.TagManager;
...

constructor(...) {
  ...
  this.tags = core.TagManager(core.TagType.KEY_VALUE, 'RFDK::SEPConfiguration');
  ...
}

private generateSpotFleetRequestConfig(props: SEPSpotFleetProps) {
  ...

  const tagSpecToken = core.Lazy.string({
    produce: () => {
      if (this.tags.hasTags()) {
        return {
          TagSpecifications: {
            ResourceType: 'spot-fleet-request',
            Tags: this.tags.render()
          }
        }
      }
      return undefined;
    }
  });
  const spotFleetRequestConfiguration = {
    ...
    TagSpecification: this.tagSpecToken
  }
  ...
}
```

??

Reply:
I switched to `TagManager` as you proposed. Also, having common tags for spot fleet requests and instances really helps to keep things simple.
If we decide to support separate tags for instances in the future, it wouldn't be hard to add an additional function.